### PR TITLE
Fixed GCC Warnings

### PR DESCRIPTION
**Pull request by AnthonyWharton**
Just rebuilding now to test, please do a clean build and check that our stuff isn't creating any more warnings.

For some reason when building clean I noticed that the irrlicht demo's returned.. can't work out why as it came from master and the last time I built master it didn't have the demo's... If you have any ideas please let me know :stuck_out_tongue:

### CODE REVIEW

**Comment by grant-m-s** [comment]
on windows they come back until you make a change that affects one of the cmake files (or sometimes when vs crashes) then they disappear no idea why

**Comment by AnthonyWharton** [comment]
Makes no sense as the option to include them is off in `extlibs/CMakeLists.txt` ....

**Comment by AnthonyWharton** [comment]
Finished the build on my machine, seems to be all good.

**Comment by grant-m-s** [comment]
:man_shrugging:

**Comment by AnthonyWharton** [comment]
Sorry missed that comment @grant-m-s .. Not sure on the status on #118 so I was leaving it for now..

File: src/client/src/system/SystemButtonRenderer.cpp [hunk 1]
```diff
@@ -200,8 +200,8 @@ void SystemButtonRenderer::updateButtons(irr::video::IVideoDriver *driver)
 				}
 			}
 			else {
-				for (int j = 0; j < buttons.size(); j++) {
-					buttons[j].triggered == true;
+				for (uint32_t j = 0; j < buttons.size(); j++) {
+					buttons[j].triggered |= true;
```
Comment (RcColes) [inline src/client/src/system/SystemButtonRenderer.cpp]: This one is an actual change, surely before it wasn't working?

Comment (AnthonyWharton) [reply src/client/src/system/SystemButtonRenderer.cpp]: This was sure to a build build warning occurring when they had just an `==`.. I can't remember the exact warning but this fixed it. 
In any case, this should have the same effect on a boolean which `buttons[j].triggered` is AFAIR.

Comment (AnthonyWharton) [reply src/client/src/system/SystemButtonRenderer.cpp]: Wait what the hell am I talking about.. One sec

Comment (AnthonyWharton) [reply src/client/src/system/SystemButtonRenderer.cpp]: Yeah I think it must have been broken before - looking into this properly now.

Comment (AnthonyWharton) [reply src/client/src/system/SystemButtonRenderer.cpp]: Ok, upon further examination and a chat with @grant-m-s, it's mean to just be a single equals sign. This is equivalent to `|=`, so to save another commit I'm going to call this good to go.

(Unless you massively object to using `|=`)

**Comment by AnthonyWharton** [comment]
Change of heart, it's rendering code and shouldn't set state - It was working with the unnecessary line before, so yup..

**Comment by jnterry** [comment]
@grant-m-s @AnthonyWharton The irrlicht examples building is controlled by a cmake option, if you run cmake-gui in the build directory you can check or uncheck the option "TEMPO_BUILD_IRRLICHT_EXTRAS". Saying that, I have no idea why it would randomly turn itself on...

Regardless, this code works, I see now warnings, merging...

**Pull request merged**